examples_tests/06.MeshLoaders/main.cpp

```diff
@@ -7,81 +7,97 @@
-
-//#include "nbl/ext/ScreenShot/ScreenShot.h"
-
+#include "../common/CommonAPI.h"
+#include "../source/Nabla/CFileSystem.h"
+#include "nbl/ext/ScreenShot/ScreenShot.h"
 
 using namespace nbl;
 using namespace core;
 
 int main()
 {
-	// create device with full flexibility over creation parameters
-	// you can add more parameters if desired, check nbl::SIrrlichtCreationParameters
-	nbl::SIrrlichtCreationParameters params;
-	params.Bits = 24; //may have to set to 32bit for some platforms
-	params.ZBufferBits = 24; //we'd like 32bit here
-	params.DriverType = video::EDT_OPENGL; //! Only Well functioning driver, software renderer left for sake of 2D image drawing
-	params.WindowSize = dimension2d<uint32_t>(1280, 720);
-	params.Fullscreen = false;
-	params.Vsync = true; //! If supported by target platform
-	params.Doublebuffer = true;
-	params.Stencilbuffer = false; //! This will not even be a choice soon
-	auto device = createDeviceEx(params);
-
-	if (!device)
-		return 1; // could not create selected driver.
-
-
-	//! disable mouse cursor, since camera will force it to the middle
-	//! and we don't want a jittery cursor in the middle distracting us
-	device->getCursorControl()->setVisible(false);
-
-	//! Since our cursor will be enslaved, there will be no way to close the window
-	//! So we listen for the "Q" key being pressed and exit the application
-	QToQuitEventReceiver receiver;
-	device->setEventReceiver(&receiver);
+    constexpr uint32_t WIN_W = 1280;
+    constexpr uint32_t WIN_H = 720;
+    constexpr uint32_t FBO_COUNT = 1u;
+
+    auto initOutput = CommonAPI::Init<WIN_W, WIN_H, FBO_COUNT>(video::EAT_OPENGL, "MeshLoaders");
+    auto window = std::move(initOutput.window);
+    auto gl = std::move(initOutput.apiConnection);
+    auto surface = std::move(initOutput.surface);
+    auto gpuPhysicalDevice = std::move(initOutput.physicalDevice);
+    auto logicalDevice = std::move(initOutput.logicalDevice);
+    auto queue = std::move(initOutput.queue);
+    auto swapchain = std::move(initOutput.swapchain);
+    auto renderpass = std::move(initOutput.renderpass);
+    auto fbo = std::move(initOutput.fbo[0]);
+    auto commandPool = std::move(initOutput.commandPool);
+
+    core::smart_refctd_ptr<nbl::video::IGPUCommandBuffer> commandBuffers[1];
+    logicalDevice->createCommandBuffers(commandPool.get(), nbl::video::IGPUCommandBuffer::EL_PRIMARY, 1, commandBuffers);
+    auto commandBuffer = commandBuffers[0];
+
+    core::smart_refctd_ptr<nbl::asset::IAssetManager> assetManager;
+    {
+        nbl::core::smart_refctd_ptr<nbl::io::IFileSystem> fileSystem = nbl::core::make_smart_refctd_ptr<nbl::io::CFileSystem>("");
+        assetManager = core::make_smart_refctd_ptr<nbl::asset::IAssetManager>(std::move(fileSystem));
+    }
 
+    core::smart_refctd_ptr<nbl::scene::ISceneManager> sceneManager;
+    {
+        // how to create properly Scene Manager in new API?
+        // only interface class provided
+    }
```

Comment:
don't do it, you only need a camera animator

Reply:
yeah, @Erfan-Ahmadi will implement it soon